### PR TITLE
Refactor run-method-style Fabric tests

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #17648, incorporating my feedback https://github.com/Lightning-AI/lightning/pull/17648#issuecomment-1555972368. 

This can serve as a demonstration of how we would like to refactor out the run-method pattern in our tests.

Part of https://github.com/Lightning-AI/lightning/issues/17479

cc @justusschock @awaelchli @carmocca @borda

### CODE REVIEW

File: tests/tests_fabric/strategies/test_deepspeed_integration.py [hunk 1]
```diff
@@ -31,78 +31,79 @@
...
+        if i == 0:
+            # save for later to check that the weights were updated
+            state_dict = deepcopy(model.state_dict())
+        optimizer.step()
+
+    # check that the model trained, the weights from step 1 do not match the weights from step 2
+    for mw_b, mw_a in zip(state_dict.values(), model.state_dict().values()):
+        assert not torch.allclose(mw_b, mw_a)
+
+    fabric.seed_everything(42)
```
Comment (Borda) [inline tests/tests_fabric/strategies/test_deepspeed_integration.py:60]: unrelated note to this PR, but shall we have seed as a wrapper to be called `with seed_...` so we can have it with model

**Review by justusschock — APPROVED** [review]

File: tests/tests_fabric/strategies/test_deepspeed_integration.py (hunk 1 above)
Comment (awaelchli) [reply tests/tests_fabric/strategies/test_deepspeed_integration.py:60]: In all cases I know, it is desired to set the seed globally for the program, but if you find a use case for it to be set locally only, feel free to open an issue or PR :)
